microcosm_postgres/dag.py

```diff
@@ -101,16 +101,39 @@ class DAGCloner(metaclass=ABCMeta):
     def __init__(self, graph):
         pass
 
-    def explain(self, **kwargs):
+    @property
+    def anonymizers(self):
+        """
+        Model-based modifying functions that remove sensitive data when cloning
+
+        """
+        return dict()
+
+    def explain(self, anonymize=False, **kwargs):
         """
         Generate a "dry run" DAG of that state that WILL be cloned.
 
         """
         root = self.retrieve_root(**kwargs)
         children = self.iter_children(root, **kwargs)
         dag = DAG.from_nodes(root, *children)
+        if anonymize:
+            dag = self.anonymize(dag)
         return self.add_edges(dag)
 
+    def anonymize(self, dag):
+        if self.anonymizers:
+            for model_name, model_nodes in dag.nodes_map.items():
+                anonymize_func = self.anonymizers.get(model_name)
+                if anonymize_func:
+                    for node in model_nodes:
+                        # disconnect node from session so it won't get modified
+                        node.store.expunge(node)
+
+                        anonymize_func(node)
```

Comment:
What's the contract here? In place modification? Return value?

Reply:
right now it's in-place (I can remove the returns if it's confusing), any preference?

Reply:
I prefer the functional style, but the overhead of reconstructing the model on edit might be too high. So unless that's easy, remove the return from the example?